cmake/modules/FindOpenGLES.cmake

```diff
@@ -1,50 +1,62 @@
-  set(OPENGLES_INCLUDE_DIR ${OPENGLES_gl_LIBRARY}/Headers)
-endif()
+                                   NO_CACHE)
+  find_path(OPENGLES_INCLUDE_DIR NAMES GLES2/gl2.h ES2/gl.h
+                                 HINTS ${PC_OPENGLES_INCLUDEDIR} ${OPENGLES_gl_LIBRARY}/Headers
+                                 NO_CACHE)
+  find_path(OPENGLES3_INCLUDE_DIR NAMES GLES3/gl3.h ES3/gl.h
+                                  HINTS ${PC_OPENGLES_INCLUDEDIR} ${OPENGLES_gl_LIBRARY}/Headers
+                                  NO_CACHE)
+
+  include(FindPackageHandleStandardArgs)
+  find_package_handle_standard_args(OpenGLES
+                                    REQUIRED_VARS OPENGLES_gl_LIBRARY OPENGLES_INCLUDE_DIR)
 
-find_path(OPENGLES3_INCLUDE_DIR GLES3/gl3.h)
+  if(OPENGLES_FOUND)
+    if(CORE_SYSTEM_NAME STREQUAL darwin_embedded)
+      # Cmake only added support for Frameworks as the IMPORTED_LOCATION as of 3.28
+      # https://gitlab.kitware.com/cmake/cmake/-/merge_requests/8586
+      # Until we move to cmake 3.28 as minimum, explicitly set to binary inside framework
+      if(OPENGLES_gl_LIBRARY MATCHES "/([^/]+)\\.framework$")
+        set(_gles_fw "${OPENGLES_gl_LIBRARY}/${CMAKE_MATCH_1}")
+        if(EXISTS "${_gles_fw}.tbd")
+          string(APPEND _gles_fw ".tbd")
+        endif()
+        set(OPENGLES_gl_LIBRARY ${_gles_fw})
+      endif()
+    endif()
 
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(OpenGLES
-                                  REQUIRED_VARS OPENGLES_gl_LIBRARY OPENGLES_INCLUDE_DIR)
+    add_library(OpenGL::GLES UNKNOWN IMPORTED)
+    set_target_properties(OpenGL::GLES PROPERTIES
+                                       IMPORTED_LOCATION "${OPENGLES_gl_LIBRARY}"
+                                       INTERFACE_INCLUDE_DIRECTORIES "${OPENGLES_INCLUDE_DIR}")
 
-find_path(OPENGLES3_INCLUDE_DIR GLES3/gl3.h
-                                PATHS ${PC_OPENGLES_INCLUDEDIR})
+    if(OPENGLES3_INCLUDE_DIR)
+      set_property(TARGET OpenGL::GLES APPEND PROPERTY
+                                       INTERFACE_INCLUDE_DIRECTORIES "${OPENGLES3_INCLUDE_DIR}")
+      set_target_properties(OpenGL::GLES PROPERTIES
+                                         INTERFACE_COMPILE_DEFINITIONS HAS_GLES=3)
+    else()
+      set_target_properties(OpenGL::GLES PROPERTIES
+                                         INTERFACE_COMPILE_DEFINITIONS HAS_GLES=2)
+    endif()
 
-if(OPENGLES_FOUND)
-  set(OPENGLES_LIBRARIES ${OPENGLES_gl_LIBRARY})
-  if(OPENGLES3_INCLUDE_DIR)
-    set(OPENGLES_INCLUDE_DIRS ${OPENGLES_INCLUDE_DIR} ${OPENGLES3_INCLUDE_DIR})
-    set(OPENGLES_DEFINITIONS -DHAS_GLES=3)
-    mark_as_advanced(OPENGLES_INCLUDE_DIR OPENGLES3_INCLUDE_DIR OPENGLES_gl_LIBRARY)
-  else()
-    set(OPENGLES_INCLUDE_DIRS ${OPENGLES_INCLUDE_DIR})
-    set(OPENGLES_DEFINITIONS -DHAS_GLES=2)
-    mark_as_advanced(OPENGLES_INCLUDE_DIR OPENGLES_gl_LIBRARY)
+    set_property(GLOBAL APPEND PROPERTY INTERNAL_DEPS_PROP OpenGL::GLES)
```

Comment:
@fuzzard: Is here missing line `set(OPENGLES_LIBRARIES ${OPENGLES_gl_LIBRARY})`?
Without it I get absolute path from build host in kodi:
```
CoreELEC VIM4:~ # ldd /usr/lib/kodi/kodi.bin
        /home/docker/build_dir/CoreELEC-21/build.CoreELEC-Amlogic-ne.aarch64-21/toolchain/aarch64-libreelec-linux-gnu/sysroot/usr/lib/libGLESv2.so => not found
```

Reply:
Short answer is no, that isnt a correct fix. the *_LIBRARIES variables (and *_DEFINITIONS, *_INCLUDE_DIRS) are a legacy way to pass info around the cmake world in our macros. Populating the target should provide all the information, of which cmake then extracts/uses as it needs behind the scenes.

At a pure blind guess im guessing some rpath bullshit, but no idea at this point in time. Grab a CMakeCache.txt from the kodi package build and we'll see whats in it

Reply:
CMakeCache.txt is 100% the same
https://gist.github.com/vpeter4/c7bfbdded5674aab37e97493d7e7763c

but in build.ninja I see that it links GLESv2 library 3 times when ok and only 1 time when not ok
diff at "Link the executable kodi.bin" block
ok https://gist.github.com/vpeter4/df53b76f8a6186f7b5dc7fa01fdb97a1
not https://gist.github.com/vpeter4/955406218b2dea5535aa3b44a5644ac3

Reply:
Its not a linker failure you're experiencing though, so the number of times it appears on the linker line for this particular issue isnt relevant as it succesfully links. What i believe its not doing is the change of rpath for the isntall step.

Cmake when it links for the build will extend out the full path of libs for that step to be able to execute from the build dir, but when it does an install its meant to filter out things like the rpath to things potentially provided for CMAKE_INSTALL_RPATH and friends.
Im not sure if its definitely the install rpath thats the issue, or potentially that theres no IMPORTED_SONAME or maybe an INSTALL_RPATH property. I havent come across a need for them as yet, so im grabbing at straws here.

Unfortunately at this point ive been unable to build libreelec as im an aarch64 host, and i get build failures from the internal glib in pkg-config. 

```
{standard input}: Assembler messages:
{standard input}:673: Error: selected processor does not support `eor3 v0.16b,v0.16b,v1.16b,v3.16b'
```

So im trying to find a reproducer without having to build the elec's at this point.

Reply:
No, it wont be that. As mentioned, cmake for the build step will use full lib paths. This is normal. When a target is installed, cmake relinks and updates/changes rpaths and other things to be installed.

Reply:
I think this issue comes from the fact that library `libGLESv2.so` doesn't have `SONAME` set (`objdump -p` doesn't show it).
There are 2 options:

1. set `SONAME` using `patchelf` binary
2. modifying cmake file
```
--- a/cmake/modules/FindOpenGLES.cmake	2023-10-07 20:07:55.000000000 +0200
+++ b/cmake/modules/FindOpenGLES.cmake	2023-10-11 15:08:10.933170795 +0200
@@ -42,9 +42,10 @@ if(NOT TARGET OpenGL::GLES)
       endif()
     endif()
 
-    add_library(OpenGL::GLES UNKNOWN IMPORTED)
+    add_library(OpenGL::GLES SHARED IMPORTED)
     set_target_properties(OpenGL::GLES PROPERTIES
                                        IMPORTED_LOCATION "${OPENGLES_gl_LIBRARY}"
+                                       IMPORTED_NO_SONAME TRUE
                                        INTERFACE_INCLUDE_DIRECTORIES "${OPENGLES_INCLUDE_DIR}")
```

I would rather see option 2 but probably needs to cover shared and static libraries.

Reply:
Definitely can be done in cmake. Can it be left as an unknown imported but just add soname property? As far as I read we might be able to add for everything, as platforms that don't use soname just disregard the propery